### PR TITLE
Add example/docs about package.js

### PR DESCRIPTION
**Pull request by litvand**
## 🌟 What is the purpose of this PR?

This PR adds a heavily commented example of `package.js`.

## 🐾 Next steps

* Documentation about `package.py`
* More detailed documentation about state and context objects
* More detailed documentation about custom loaders and getters

## 🔍 What does this change?

* Adds a file in a new `docs` directory

### Does this require a change to the docs?

No.

## 🔗 Related links

[Asana task](https://app.asana.com/0/1199550852792314/1201481006654706/f) (_internal_)

### CODE REVIEW

**Comment by nonparibus** [comment]
So we're planning to include dev docs inline within the folder for now?

Are you thinking we'll direct users to this from the forthcoming https://hash.dev/ site for our open-source and fair-code projects (initially HASH and hEngine) or will this be moved out of the repo entirely later?

File: packages/engine/docs/example-package.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+///
+const start_sim = (experiment, sim, init_message, init_context) => {
+    sim.size = experiment.min_size + init_message.additional_size
+    return {
+        "warnings": ["user warning"],
+        "errors": ["user error"],
+        "loaders": {
+            "column_name": hash_util.load_shallow,
+            "another_column": hash_util.load_full
+        }
```
Comment (TimDiekmann) [inline packages/engine/docs/example-package.js:101]:
```suggestion
        },
```

**Comment by Alfred-Mountfield** [comment]
> So we're planning to include dev docs inline within the folder for now?
> 
> Are you thinking we'll direct users to this from the forthcoming https://hash.dev/ site for our open-source and fair-code projects (initially HASH and hEngine) or will this be moved out of the repo entirely later?

Having just had a discussion in stand-up, we think it makes sense to do something in-between that we'll figure out when the site is live and a structure is decided. By that I mean that we'll still want to generate some Rust docs* and link to those from the HASH site, but we'll also want to have simulation guidance, perhaps hEngine self-hosting guidance, etc. on the site itself rather than in the Rust docs.

We need to figure out what to do with things specific to the code implementation like the contents of this PR, we might want to find a way to integrate it in the generated Rust-docs.

* Like [this](https://docs.rs/flatbuffers/2.0.0/flatbuffers/) although perhaps we will self-host if we don't want to publicise the module, that can be discussed later.

**Review by Alfred-Mountfield — DISMISSED** [review]
We should look at moving this into an .rs file later on when we come to actually generating our rust documentation for the engine, so that it's in the same place.

**Review by TimDiekmann — DISMISSED** [review]

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**